pkg/logql/log/metrics_extraction.go

```diff
@@ -80,22 +80,25 @@ type streamLineSampleExtractor struct {
 	builder *LabelsBuilder
 }
 
-func (l *streamLineSampleExtractor) Process(ts int64, line []byte) (float64, LabelsResult, bool) {
+func (l *streamLineSampleExtractor) Process(ts int64, line []byte, metadataLabels ...labels.Label) (float64, LabelsResult, bool) {
+	l.builder.Reset()
```

Comment:
it changes the logic. previously if stage is NoopStage we `return l.LineExtractor(line), l.builder.GroupedLabels(), true` that uses values from the builder, otherwise we rest it and continue processing

Reply:
If I read the code correctly, it should not cause any problems because `Reset` just clears any buffered changes done on top of base labels.

Reply:
> [Reset](https://github.com/grafana/loki/blob/23e59fa4a95b64d2a77ac6377f287938ed363872/pkg/logql/log/labels.go#L140-L144) just clears any buffered changes done on top of base labels.

Correct, that's my understanding as well